core/controllers/editor.py

```diff
@@ -825,8 +823,16 @@ class ExplorationStatisticsHandler(EditorHandler):
 
     GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
 
+    @acl_decorators.can_view_exploration_stats
     def get(self, exploration_id, exploration_version):
         """Handles GET requests."""
+        # This check is needed in order to show the correct page when a 404
```

Comment:
I don't think this is needed anywhere except the page that serves the exploration to be played. No other pages are meant to be served via iframe.